livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterprereter.java

```diff
@@ -422,6 +423,8 @@ private String callRestAPI(String targetURL, String method, String jsonData)
         if (cause.getResponseBodyAsString().matches(SESSION_NOT_FOUND_PATTERN)) {
           throw new SessionNotFoundException(cause.getResponseBodyAsString());
         }
+        throw new LivyException(cause.getResponseBodyAsString() + "\n"
+            + ExceptionUtils.getFullStackTrace(ExceptionUtils.getRootCause(e)));
```

Comment:
I'm not sure I get it though. Here we are creating a new exception with the full stack trace of the inner/original exception in the exception message itself. In the example you point to `setResult(e.getMessage())` would then get the whole message + stack and set that as the result of the paragraph, which would be huge?